minimizer/minimize.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if ! [[ -x "$(command -v dustmite)" ]]; then
+    echo "No DustMite installed. It is typically contained in a package called 'dtools'."
+    exit 1
+fi
+
+available_scripts=()
+for filename in ./conditions/*.sh; do
+    filename=$(basename -- "$filename")
+    filename="${filename%.*}"
+    available_scripts+=("$filename")
+done
+
+if [[ $# -lt 2 ]]; then
+    echo "usage: $0 <file> <condition>"
+    echo "available conditions are:"
+    printf '\t%s\n' "${available_scripts[@]}"
+    exit 1
+fi
+
+FILE="$1"
+
+if ! test -f "$FILE"; then
+    echo "Input file '$FILE' not found."
+    exit 1
+fi
+
+SCRIPT_PATH=$(cd "$(dirname "${BASH_SOURCE[0]}")" || exit; pwd -P)
+CONDITION_SCRIPT="$2"
+
+if ! test -f "$SCRIPT_PATH/conditions/$CONDITION_SCRIPT.sh"; then
+    echo "Condition script '$SCRIPT_PATH/conditions/$CONDITION_SCRIPT.sh' not found."
+    echo "available conditions are:"
+    printf '\t%s\n' "${available_scripts[@]}"
+    exit 1
+fi
+
+shift
+shift
+CONDITION="$*"
+WORKSPACE=$SCRIPT_PATH/workspace
+mkdir "$WORKSPACE"
+ln -s "$(realpath "$FILE")" "$WORKSPACE"/input.c
+
+dustmite --strip-comments --split "*.c:d" "$WORKSPACE"/input.c "$SCRIPT_PATH/conditions/$CONDITION_SCRIPT.sh $CONDITION"
```

Comment:
Hmm ... this doesn't work if I quote the condition:
```
./minimize.sh ../crash.c output_contains 'stack overflow'
+ dustmite --strip-comments --split '*.c:d' /home/joshua/src/rust/rcc/minimizer/workspace/input.c '/home/joshua/src/rust/rcc/minimizer/conditions/output_contains.sh stack overflow'
Loading /home/joshua/src/rust/rcc/minimizer/workspace/input.c
None => No
object.Exception@dustmite.d(270): Initial test fails (try using --no-redirect for details)
```
I tried `"$@"`, `"$*"`, and `$*` and all of them failed one way or another. The issue is that dustmite requires the command to be a single argument instead of multiple, normally `"$@"` would work for this but it creates multiple arguments instead of one. The awful hack is

```suggestion
dustmite --strip-comments --split "*.c:d" "$WORKSPACE"/input.c "$(printf " %q" "$SCRIPT_PATH/conditions/$CONDITION_SCRIPT.sh" "$@")"
```

This preserves the quoting behavior even if the original string passed by the user had spaces or quotes. See https://stackoverflow.com/questions/10835933/preserve-quotes-in-bash-arguments.